### PR TITLE
soc: silabs_exx32: Enable mpu on efr32mg soc

### PR DESCRIPTION
**Pull request by chrta**
Enables the arm v7m mpu on the efr32mg soc and the board
efr32mg_sltb004a.

Tested on hardware with samples/mpu/mpu_test and
tests/kernel/mem_protect

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
Generally looks good, nit on the .dts update and whitespace formatting.

**Review by ioannisg — CHANGES_REQUESTED** [review]
Looks good - only one comment regarding cleaning up DTS

BTW, are you able, now to execute all sanity check `-t usermode` tests properly?

**Review by dbkinder — APPROVED** [review]

**Comment by chrta** [comment]
@ioannisg 
> BTW, are you able, now to execute all sanity check -t usermode tests properly?

How do i do that?
I tried executing `scripts/sanitycheck --device-testing --device-serial /dev/ttyACM0 -p efr32mg_sltb004a` but this broke my onboard debug adapter (segger) somehow.

File: dts/arm/silabs/efr32mg.dtsi [hunk 1]
```diff
@@ -13,6 +13,14 @@
...
 			compatible = "arm,cortex-m4f";
 			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			
+			mpu: mpu@e000ed90 {
+                                compatible = "arm,armv7m-mpu";
+                                reg = <0xe000ed90 0x40>;
+                                arm,num-mpu-regions = <8>;
+                        };
```
[the remark this answers is not in the document]
Comment (chrta) [reply dts/arm/silabs/efr32mg.dtsi]: done (removed changes to this file)

File: dts/arm/silabs/efr32mg.dtsi [hunk 2]
```diff
@@ -13,6 +13,14 @@
 		cpu0: cpu@0 {
 			compatible = "arm,cortex-m4f";
 			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			
+			mpu: mpu@e000ed90 {
```
[the remark this answers is not in the document]
Comment (chrta) [reply dts/arm/silabs/efr32mg.dtsi]: thanks for the info, done.

**Comment by ioannisg** [comment]
> @ioannisg
> 
> > BTW, are you able, now to execute all sanity check -t usermode tests properly?
> 
> How do i do that?
> I tried executing `scripts/sanitycheck --device-testing --device-serial /dev/ttyACM0 -p efr32mg_sltb004a` ~but this broke my onboard debug adapter (segger) somehow~.

Just confirm that you are able to run tests on the platform that are tagged with `userspace`.

**Review by ioannisg — APPROVED** [review]

**Comment by chrta** [comment]
Just for reference: I executed
```scripts/sanitycheck --device-testing --device-serial /dev/ttyACM0 --west-flash="--reset-after-load" -p efr32mg_sltb004a -t userspace -v```
It returned:
```
40 of 42 tests passed (100.00%), 0 failed, 2 skipped with 0 warnings in 357.63 seconds
```

**Comment by ioannisg** [comment]
> Just for reference: I executed
> `scripts/sanitycheck --device-testing --device-serial /dev/ttyACM0 --west-flash="--reset-after-load" -p efr32mg_sltb004a -t userspace -v`
> It returned:
> 
> ```
> 40 of 42 tests passed (100.00%), 0 failed, 2 skipped with 0 warnings in 357.63 seconds
> ```

Then that should be fine for me to merge. I leave this to @galak

File: soc/arm/silabs_exx32/efr32mg12p/dts_fixup.h [hunk 3]
```diff
@@ -9,6 +9,9 @@
 /* SoC level DTS fixup file */
 
 #define DT_NUM_IRQ_PRIO_BITS	DT_ARM_V7M_NVIC_E000E100_ARM_NUM_IRQ_PRIORITY_BITS
+
+#define DT_NUM_MPU_REGIONS      DT_ARM_ARMV7M_MPU_E000ED90_ARM_NUM_MPU_REGIONS
```
Comment (ioannisg) [inline soc/arm/silabs_exx32/efr32mg12p/dts_fixup.h]: This must be removed, cause the DT_ define is not generated

**Review by ioannisg — CHANGES_REQUESTED** [review]
I think the DT_ define for MPU regions is not generated any more so the fixup macro should be removed

File: soc/arm/silabs_exx32/efr32mg12p/dts_fixup.h (hunk 3 above)
Comment (chrta) [reply soc/arm/silabs_exx32/efr32mg12p/dts_fixup.h]: done and retested, the sanitycheck tests still pass.

**Review by ioannisg — APPROVED** [review]

**Comment by ioannisg** [comment]
@galak ping for review

**Review by galak — APPROVED** [review]

**Pull request merged**